crates/handlers/src/graphql/mutations/user.rs

```diff
@@ -585,9 +588,16 @@ impl UserMutations {
             return Ok(UnlockUserPayload::NotFound);
         };
 
-        // Call the homeserver synchronously to unlock the user
```

Comment:
Maybe this should have the 'old' behaviour of reactivating. This is an internal API, and no one should be relying on this except us, so to avoid breaking our internal tooling, I would not introduce new flags, and make this API reactivate users properly

Reply:
Completely reasonable!  Done with 43806cdf (along with a slight update to documentation)